spec/controllers/admin/product_import_controller_spec.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Admin::ProductImportController, type: :controller do
+  describe 'tmp_file_path_regex' do
+    let!(:regex){ described_class::TEPM_FILE_PATH_REGEX }
+    context 'file extension' do
+      it 'should authorize csv extension' do
+        path = '/tmp/product_import123/import.csv'
+        expect(regex.match?(path)).to be_truthy
+      end
+
+      it 'should reject other extensions' do
+        path = '/tmp/product_import123/import.pdf'
+        expect(regex.match?(path)).to be_falsey
+        path1 = '/tmp/product_import123/import.xslx'
+        expect(regex.match?(path1)).to be_falsey
+      end
+    end
+
+    context 'folder path' do
+      it 'should authorize valid paths' do
+        path = '/tmp/product_import123/import.csv'
+        expect(regex.match?(path)).to be_truthy
+        path1 = '/tmp/product_importabc/import.csv'
+        expect(regex.match?(path1)).to be_truthy
+        path2 = '/tmp/product_importABC-abc-123/import.csv'
+        expect(regex.match?(path2)).to be_truthy
+      end
+
+      it 'should reject invalid paths' do
+        path = '/tmp/product_import123/../etc/import.csv'
+        expect(regex.match?(path)).to be_falsey
+
+        path1 = '/tmp/product_import../etc/import.csv'
+        expect(regex.match?(path1)).to be_falsey
+
+        path2 = '/tmp/product_import132%2F..%2Fetc%2F/import.csv'
+        expect(regex.match?(path2)).to be_falsey
+
+        path3 = '/etc/tmp/product_import123/import.csv'
+        expect(regex.match?(path3)).to be_falsey
+      end
+    end
+  end
+
+  describe 'sanitize_file_path' do
+    let(:folder_path){ '/tmp/product_import123' }
+    let(:file_path) { "#{folder_path}/import.csv" }
+
+    before do
+      FileUtils.mkdir_p(folder_path)
+      File.new(file_path, 'w') unless File.exist?(file_path)
+    end
+
+    def subject(path)
+      # expose the private method sanitize_file_path
+      described_class.class_eval { public :sanitize_file_path }
+      described_class.new.__send__(:sanitize_file_path, path)
+    end
```

Comment:
If this was in a service object, you would have a public interface to test with.

Reply:
done